tests/integration/docker/dns.bats

```diff
@@ -0,0 +1,100 @@
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# DNS tests using Swarm
+
+SRC="${BATS_TEST_DIRNAME}/../../lib/"
+# number of replicas to launch
+number_of_replicas=1
+# total number of replicas to launch with nginx
+total_replicas=2
+# number of attempts
+number_of_attempts=5
+
+#This function will verify that swarm is not running or it finishes
+function clean_swarm_status() {
+	for j in `seq 0 $number_of_attempts`; do
+		if $DOCKER_EXE node ls; then
+			$DOCKER_EXE swarm leave --force
+			# "docker swarm leave" is not immediate so it requires some time to finish
+			sleep 5
+		else
+			break
+		fi
+	done
+}
+
+setup() {
+	source $SRC/test-common.bash
+	runtime_docker
+	clean_swarm_status
+	interfaces=($(readlink /sys/class/net/* | grep -i pci | xargs basename -a))
+	swarm_interface_arg=""
+	for i in ${interfaces[@]}; do
+		if [ "`cat /sys/class/net/${i}/operstate`" == "up" ]; then
+			swarm_interface_arg="--advertise-addr ${i}"
+			break;
+		fi
+	done
+	$DOCKER_EXE swarm init ${swarm_interface_arg}
+	$DOCKER_EXE service create --name testswarm1 --replicas $number_of_replicas --publish 8080:80 nginx /bin/bash -c "hostname > /usr/share/nginx/html/hostname; nginx -g \"daemon off;\""
+	$DOCKER_EXE service create --name testswarm2 --replicas $number_of_replicas --publish 8082:80 nginx /bin/bash -c "hostname > /usr/share/nginx/html/hostname; nginx -g \"daemon off;\""	
+	$DOCKER_EXE service create --name testdns --replicas $number_of_replicas --publish 8084:80 mcastelino/nettools sleep 60000
+	while [ `$DOCKER_EXE ps --filter status=running --filter ancestor=nginx:latest -q | wc -l` -lt $total_replicas ]; do
+		sleep 1
+	done
+	while [ `$DOCKER_EXE ps --filter status=running --filter ancestor=mcastelino/nettools:latest -q | wc -l` -lt $number_of_replicas ]; do
+		sleep 1
+	done
+}
+
+@test "DNS test to check Server and IP Address" {
+	skip "this is not working (this is related with https://github.com/01org/cc-oci-runtime/issues/578)"
+	container_id_dns=`$DOCKER_EXE ps -qaf "name=testdns"`
+	expected_virtual_ip_1=`$DOCKER_EXE service inspect testswarm1 -f '{{if eq 1 (len .Endpoint.VirtualIPs)}}{{with index .Endpoint.VirtualIPs 0}}{{.Addr}}{{end}}{{end}}' | cut -d "/" -f1`
+	$DOCKER_EXE exec -it $container_id_dns nslookup testswarm1 | grep Address | tail -1 | cut -d ':' -f2 | tr -d ' ' > ip_1
+	sed 's/\r//g' ip_1 > final_ip_1
```

Comment:
You cannot assume that the current directory is writable. This (and all new redirects below) needs to be something like:

```
final_ip_1=$(mktemp)
sed 's/\r//g' ip_1 > "$final_ip_1"
```